src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
@KjellKod @john-gress we can discuss it when i get back, but at home I added boost serialization to my personal library to skirt around this issue. as long as you pass in a serializable class it will serialize, allocate the msg, send, receive, reconstruct obj, deallocate msg.

not sure what the approach we want to take here is.